connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -514,6 +541,40 @@ private void updateTopicConfigs(Map<String, Config> topicConfigs) {
         }));
     }
 
+    // visible for testing
+    @SuppressWarnings("deprecation")
+    void incrementalAlterConfigs(Map<String, Config> topicConfigs) {
+        Map<ConfigResource, Collection<AlterConfigOp>> configOps = new HashMap<>();
+        for (Map.Entry<String, Config> topicConfig : topicConfigs.entrySet()) {
+            Collection<AlterConfigOp> ops = new ArrayList<>();
+            ConfigResource configResource = new ConfigResource(ConfigResource.Type.TOPIC, topicConfig.getKey());
+            for (ConfigEntry config : topicConfig.getValue().entries()) {
+                if (config.isDefault() && !shouldReplicateSourceDefault(config.name())) {
+                    ops.add(new AlterConfigOp(config, AlterConfigOp.OpType.DELETE));
+                } else {
+                    ops.add(new AlterConfigOp(config, AlterConfigOp.OpType.SET));
+                }
+            }
+            configOps.put(configResource, ops);
+        }
+        log.trace("Syncing configs for {} topics.", configOps.size());
+        targetAdminClient.incrementalAlterConfigs(configOps).values().forEach((k, v) -> v.whenComplete((x, e) -> {
+            if (e != null) {
+                if (useIncrementalAlterConfigs.equals(MirrorSourceConfig.REQUEST_INCREMENTAL_ALTER_CONFIG)
+                        && e instanceof UnsupportedVersionException) {
+                    //Fallback logic
+                    log.warn("The target cluster {} is not compatible with IncrementalAlterConfigs API. Therefore using deprecated AlterConfigs API for syncing topic configurations", sourceAndTarget.target(), e);
+                    useIncrementalAlterConfigs = MirrorSourceConfig.NEVER_USE_INCREMENTAL_ALTER_CONFIG;
+                } else if (useIncrementalAlterConfigs.equals(MirrorSourceConfig.REQUIRE_INCREMENTAL_ALTER_CONFIG)
+                        && e instanceof UnsupportedVersionException) {
+                    context.raiseError(new UnsupportedVersionException("use.incremental.alter.configs was set to \"required\" however the target cluster is not compatible with IncrementalAlterConfigs API"));
```

Comment:
Two small things:

1. We should log an ERROR level message when this fails (it will be redundant in most cases, but it could come in handy in case the call to `ConnectorContext::raiseError` fails)
2. We should include the original `UnsupportedVersionException` in case there's other useful information in that exception's stack trace and/or message

```suggestion
                    log.error("Failed to sync configs for topic {} on cluster {} with IncrementalAlterConfigs API", k.name(), sourceAndTarget.target(), e);
                    context.raiseError(new ConnectException(
                            "use.incremental.alter.configs was set to \"required\", but the target cluster '"
                                    + sourceAndTarget.target() + "' is not compatible with IncrementalAlterConfigs API",
                            e
                    ));
```